### PR TITLE
Removing LoggerMetricsPublisher

### PR DESCRIPTION
**Pull request by danielpassos**
There is no sense to have `LoggerMetricsPublisher` anymore since [`MobileCore`](https://github.com/aerogear/aerogear-android-sdk/blob/master/core/src/main/java/org/aerogear/mobile/core/MobileCore.java) is going to [raise](https://github.com/aerogear/aerogear-android-sdk/blob/master/core/src/main/java/org/aerogear/mobile/core/MobileCore.java#L164-L167) a [ConfigurationNotFoundException](https://github.com/aerogear/aerogear-android-sdk/blob/master/core/src/main/java/org/aerogear/mobile/core/exception/ConfigurationNotFoundException.java) is there is no _metrics_ entry in `mobile-service.json` and someone try to use [`MetricsService`](https://github.com/aerogear/aerogear-android-sdk/blob/master/core/src/main/java/org/aerogear/mobile/core/metrics/MetricsService.java)

### CODE REVIEW

**Comment by wtrocki** [comment]
`NetworkMetricsPublisher` is not protected for null url and it is crashing with `NullPointerException`

**Comment by wtrocki** [comment]
Ignore my previous comment. 
If there is no URL we probably do not want to collect metrics at all.

**Comment by danielpassos** [comment]
@wtrocki I have added some SanityCheck to make you happy :kissing_heart:

**Comment by secondsun** [comment]
I think that logger metrics are good for debugging, but we might want to think about how we enable / disable it instead.

File: core/src/main/java/org/aerogear/mobile/core/metrics/MetricsPublisher.java [hunk 1]
```diff
@@ -24,7 +24,7 @@
     private final Metrics[] defaultMetrics;
 
     public MetricsPublisher(final Context context) {
-        this.context = context;
+        this.context = nonNull(context, "context");
```
Comment (ziccardi) [inline core/src/main/java/org/aerogear/mobile/core/metrics/MetricsPublisher.java]: This should be `this.context = nonNull(context, "context").getApplicationContext();` for safety

**Comment by wtrocki** [comment]
@danielpassos This looks like forgotten PR :-) However we have removed that everywhere else. @ziccardi @secondsun are you ok to get that merged?

**Comment by ziccardi** [comment]
@wtrocki : other than the comment I already left on this PR, LGTM

**Comment by danielpassos** [comment]
@wtrocki How can I simulate the NPE problem?

**Pull request merged**